programs/bpf_loader/src/lib.rs

```diff
@@ -195,6 +199,27 @@ macro_rules! deploy_program {
     };
 }
 
+// This function/wrapper is added specifically for non "svm-internal" users. It enables the reducing
+// of public visibility of some internal APIs.
+pub fn deploy_program(
+    invoke_context: &mut InvokeContext,
```

Comment:
This won't work and is the reason why `deploy_program!` is a macro and not a function in the first place. Because the subsequent fields are often borrowed from accounts in the transaction, thus borrowed from `invoke_context`. And these borrows would collide.

Reply:
Also, we could expose `deploy_program_internal()` and simply rename it `deploy_program()`.

Reply:
This new function is being used only by the `core_bpf_migration` code, which will eventually go away. So the API will be `agave-internal` and eventually deleted.

Maybe renaming `deploy_program_internal()` an using it will be better approach. I'll investigate.